platforms/core-runtime/build-operations-trace/src/main/java/org/gradle/internal/operations/trace/SerializedOperationProgress.java

```diff
@@ -41,7 +41,7 @@ class SerializedOperationProgress implements SerializedOperation {
 
     SerializedOperationProgress(Map<String, ?> map) {
         this.id = ((Integer) map.get("id")).longValue();
-        this.time = (Long) map.get("time");
+        this.time = map.get("time") instanceof Integer ? -1L : (Long) map.get("time"); // JSON serialization read the -1 sentinel value as an Int
```

Comment:
I don't think the time would ever be small enough to fit in an `Integer` though, since it's time since the epoch and we're not using historical data.  But using `Number` is a better solution, I'll update this.